yjit/src/options.rs

```diff
@@ -47,40 +47,40 @@ macro_rules! get_option {
 }
 pub(crate) use get_option;
 
-pub fn parse_option(str_ptr: *const std::os::raw::c_char) -> bool
+/// Expected to receive what comes after the third dash in "--yjit-*".
+/// Empty string means user passed only "--yjit". C code rejects when
+/// they pass exact "--yjit-".
+pub fn parse_option(str_ptr: *const std::os::raw::c_char) -> Option<()>
 {
     let c_str: &CStr = unsafe { CStr::from_ptr(str_ptr) };
-    let str_slice: &str = c_str.to_str().unwrap();
-    let opt_str: String = str_slice.to_owned();
+    let opt_str: &str = c_str.to_str().ok()?;
```

Comment:
This is the last two points from the commit message.
  - Return parse failure instead of panicking when input is not utf8
   - Remove unnecessary `str.to_owned()` which allocates and copies the
     string.
In case you are not familiar with the question mark operator: https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html#a-shortcut-for-propagating-errors-the--operator basically sugar for check and early return.

`.ok` converts Result into an Option, dropping the error type. This is the nicest way to drop the error value I found, also the reason why the return type is `Option<()>` and not `Result<(), ()>` since this plays nicer with `?`.